Feature/claims 3 1 improvements

Kristinn-Stefansson opened the pull request:
## [Claims 3.2] - 2025-11-20

- Point release 3.2 of TS 316 after further updates by workgroup 7, based on lessons from Claims 3.1 implementation.
- Implements the changes added to PR #198 after release of Claims version 3.1, with references to issues:
  - #254 Settings object missing in GET /v1/claimtemplates/{claimTemplateId}
  - #252 Inconsistent parameter naming
  - #249 Consider introducing a shared ClaimBase schema
  - #247 Improved batch handling
  - #246 Set correct nullable fields in claim schema (PATCH)
  - #244 ExpirationDate should be required when creating claims (single and batch)
  - #242 Simplify history response, use lighter object instead of ClaimDetails
  - #241 Return both print and directPaymentClaim fields when creating a claim

## [Claims 3.1] - 2025-03-19

- Point release of TS 316 after updates by workgroup 7.
- Implements the changes in PR #198, with references to issues:
  - #194 Fixed the description for default interests (#194)
  - #182 Changed all amount properties that were defined as integers to number
  - #183 Renamed isPermitOutOfSequencePayment to isOutOfSequencePaymentAllowed
  - #185 dateReference to referenceDate
  - #186 Payment batch id in claim transaction 
  - #191 PaymentFee terminology
  - #192 Feature/include claim type
  - #184 Renamed amount properties using *Amount and *AmountDue postfixes
  - #187 Renamed a few properties in claimTransaction to be more descriptive
  - #193 Creation date included in caimDetails
  - #181 Claims - Use different schemas for batch initiation
  - #199 Renamed lastChanged property
  - #200 Removed unnecessary properties from defaultInterest
  - #206 Remove IsPostRefDate
  - #205 Remove DueDateRate from CurrencyPaymentRate
  - #204 Remove Claimant Address from printing object
  - #220 Fix examples
  - #218 Fix #215 - EnumType 
  - #217 Feature/fix214 description
  - #221 Fix #216 missing references
  - #232 Feature/update316and315

Summary of Overall Changes in 3.1

The PR is the result discussions and comments were made regarding various aspects of the IOBWS 3.0 specification. The following changes that have been proposed and accepted and form the proposal for the IOBWS 3.1. are summarized below. For details see the PRs merged into the underlying branch, comments and issues that are referenced:

1. Property Name Changes: Descriptive property names were suggested and accepted for clarity and consistency. This includes renaming properties in `claimDetails`, `claimTransaction`, and other related objects to provide clearer understanding of their purpose and meaning.

2. Additional ClaimType Property: A new property called `ClaimType` was proposed and accepted for inclusion in the response data of the `GetClaims` API endpoint. This property indicates the type of the claim, such as normal claim, optional payment claim, refund claim, or government balance claim. This addition helps differentiate between different types of claims in the response.

3. Payment Properties Terminology Refinement: The terminology used for payment properties was refined to improve clarity and accuracy. Changes include replacing "directDebitFee" with "paperlessFee" and aligning the naming with the RB service backend terminology for printing fees and paperless fees.

4. CreatedDate Property Addition: A new property called `CreatedDate` was proposed and accepted to provide information about the creation date of each claim in the `GetClaims` response. This addition facilitates data analysis and processing by including the creation date information.

5. Revisiting Mixed Batches: The need for supporting mixed batches, where create, alter, and cancel operations are combined, was discussed. It was concluded that currently, there is no business requirement for mixed batches. The PR submitted separates creation and updates in batches, and future considerations for supporting mixed batches will depend on the roadmap decisions of the IH system at RB.

6. The defaultInterest entity included enums with only one option, giving the illusion of choice.The properties in question were calculationPeriod, calculationDayRule, and defaultInterestException. These where remove from the IOBWS 3.0 specification.

7. Removal of isPostRefDate: This option was present in claimDetails and createClaimDetails, but no longer supported by the underlying core systems due to lack of use.

8. Removal of DueDateRate: This was another feature no longer present in the underlying core system in the current form that was removed to prevent unnecessary confusion.


The changes overall aim to improve clarity, consistency, and functionality within the IOBWS 3.0 specification. They address naming conventions, property additions, and refine terminology to enhance understanding and usability for implementers and users of the API.

danielsveins committed:
Fixed the description for default interests (#194)

Co-authored-by: Daníel Sveinsson <daniel.sveinsson@arionbanki.is>

danielsveins committed:
Changed all amount properties that were defined as integers to numbers (#182)

Co-authored-by: Daníel Sveinsson <daniel.sveinsson@arionbanki.is>

danielsveins committed:
Renamed isPermitOutOfSequencePayment to isOutOfSequencePaymentAllowed (#183)

Co-authored-by: Daníel Sveinsson <daniel.sveinsson@arionbanki.is>

danielsveins committed:
Renamed dateReference to referenceDate (#185)

Co-authored-by: Daníel Sveinsson <daniel.sveinsson@arionbanki.is>

danielsveins committed:
Payment batch id in claim transaction (#186)

* Added paymentBatchId to claimTransaction instead of batchId

* Renamed properties in claimTransaction

---------

Co-authored-by: Daníel Sveinsson <daniel.sveinsson@arionbanki.is>

danielsveins committed:
PaymentFee terminology (#191)

* Renamed paymentFee properties

* Removed *Amount postfix in accordance to agreed upon naming in issue #178

---------

Co-authored-by: Daníel Sveinsson <daniel.sveinsson@arionbanki.is>

danielsveins committed:
Feature/include claim type (#192)

* Added claim type to claimDetails and createClaimDetails

* Set default value for claimType

---------

Co-authored-by: Daníel Sveinsson <daniel.sveinsson@arionbanki.is>

danielsveins committed:
Renamed amount properties using *Amount and *AmountDue postfixes (#184)

Co-authored-by: Daníel Sveinsson <daniel.sveinsson@arionbanki.is>

danielsveins committed:
Renamed a few properties in claimTransaction to be more descriptive (#187)

Co-authored-by: Daníel Sveinsson <daniel.sveinsson@arionbanki.is>

danielsveins committed:
Creation date included in caimDetails (#193)

* CreatedDate included in caimDetails

* Changed the description of createdDate

---------

Co-authored-by: Daníel Sveinsson <daniel.sveinsson@arionbanki.is>

haukurhrafn committed:
Claims - Use different schemas for batch initiation (#181)

* use diffrent schemas for batch initiation

* update batch description text

* use alter instead of merge

* Added actionType to batchInitiation and specified it as type descriminator to aid with deserialization.

asgeirhall commented:
Just wanted to point out that LastChanged is not consistent with other dates in the schema.

        - "DueDate"
        - "FinalDueDate"
        - "ExpirationDate"
        - "ClosingDate"
        - "CreationDate"
        - "LastChanged"

danielsveins commented:
@asgeirhall good catch, we'll change it to _LastChangeDate_ for consistency.

danielsveins committed:
Renamed lastChanged property (#199)

* Renamed lastChanged to lastChangeDate

* Updated description of lastChangeDate

---------

Co-authored-by: Daníel Sveinsson <daniel.sveinsson@arionbanki.is>

danielsveins committed:
Removed unnecessary properties from defaultInterest (#200)

Co-authored-by: Daníel Sveinsson <daniel.sveinsson@arionbanki.is>

thebingi committed:
Remove IsPostRefDate (#206)

See acceptance on issue.

thebingi committed:
Remove DueDateRate from CurrencyPaymentRate (#205)

See acceptance on issue #202

thebingi committed:
Remove Claimant Address from printing object (#204)

thebingi committed:
The claimId should be included exclusively in the URI to adhere to RESTful principles, which dictate that the URI should uniquely identify the resource to be modified. The request body should only contain the fields that are intended to be updated. (#211)

Meeting TN-FMÞ-VH7-21 reviewed at this and was unanimous in this being an improvement that could be handled as a patch to the 3.1 branch that would not affect any current implementations, there being none at the moment.

asgeirhall commented:
Nust noticed the version of the file is still 3.0 should it not be 3.1?
title: IOBWS 3.0 Domestic Claims
version: "3.0"

asgeirhall commented on Deliverables/IOBWS-Claims3.0.yaml:
```diff
@@ -1495,27 +1455,38 @@ components:
         Merge patch claim object
       type: object
-      properties:
-        claimKey:
```
Why was claimKey removed from mergeClaimDetails?   How are we going to reference the claim in the merge?

asgeirhall replied:
I would like to repeat this issue,  I understand why it was removed from single claim update.  
**BUT its also used for alterClaimBatchDetails and there its must to have claimId reference**

thebingi committed:
Fix examples (#220)

Meeting TN-FMÞ-VH7-22 reviewed the status of this item, accepted for merge.

thebingi committed:
Fix #215 - EnumType (#218)

Meeting TN-FMÞ-VH7-24 reviewed the status of this item, was accepted for merge

thebingi committed:
Feature/fix214 description (#217)

* The claimId should be included exclusively in the URI to adhere to RESTful principles, which dictate that the URI should uniquely identify the resource to be modified. The request body should only contain the fields that are intended to be updated.

* Fix description

doddi321 commented on Deliverables/IOBWS-Claims3.0.yaml:
```diff
@@ -2961,73 +2961,62 @@ components:
             "referenceDate": "FinalDueDate",
             "first": {
               "chargeType": "Percentage",
               "day": 0,
```
does it make sense to rename this to "days"? Seems more descriptive for what it is.

doddi321 commented on Deliverables/IOBWS-Claims3.0.yaml:
```diff
@@ -1120,19 +1122,18 @@ components:
       required:
-        - paymentFee
-        - directDebitFee
+        - printingFee
```
Also for consistency what you seem to be doing in this PR, rename this to printingFeeAmount and paperlessFeeAmount?  (or directDebitFeeAmount and paymentFeeAmount)

thebingi committed:
Fix #216 missing references (#221)

TN-FMÞ-VH7-27 reviewed the status and on basis of the feedback this was merged.

* Fix #216 missing references

* Minir fix

* Another minor change

kristinnstefansson committed:
Feature/update316and315 (#232)

* Rename Claims to 3.1 to reflect update
  Also updated Claim 3.1 in reference
* Add 315 (originally by copying 314)
* Update layout adjustments and participant details in 315 documents

---------

Co-authored-by: Björn Björnsson <bjorn.ingi.bjornsson@arionbanki.is>

kristinnstefansson committed:
Update 315 rendition

kristinnstefansson committed:
Update CHANGELOG and documentation for Claims 3.1 release

kristinnstefansson committed:
Update Claims specification to version 3.1 in the Technical Specification document

kristinnstefansson committed:
Merge branch 'master' into feature/claims_3_1_improvements

kristinnstefansson committed:
Merge remote-tracking branch 'origin/master' into feature/claims_3_1_improvements

kristinnstefansson committed:
Update ÍST TS 315 and 316 documents with revised entry into force dates and funding details

kristinnstefansson committed:
Update ÍST TS 316 Technical Guidelines with revised scope details

asgeirhall commented:
Is alterClaimBatchDetails not missing claimKey property?

kristinnstefansson committed:
Merge remote-tracking branch 'origin/master' into feature/claims_3_1_improvements

thebingi committed:
Print  also returned (#255)

Approved after review in meeting TN-FMÞ-VH7-36 on the 9.10.2025.

thebingi committed:
Settings added to claimTemplateDetails (#253)

Approved after review in meeting TN-FMÞ-VH7-36 on the 9.10.2025.

thebingi committed:
Improved batch handling (#251)

Approved after review in meeting TN-FMÞ-VH7-36 on the 9.10.2025. 

* Update /v1/batches docs to include five initiation types, require full claim objects for modifications, and extend actionType with new values and mappings.

* alterClaimDetails object added for alterBatch

* CancelBatch added

* SecondaryCollection for Batch added

* return secondaryCollection added

* Examples added and updated

thebingi committed:
Refactor claimHistoryEvent to use claimEvent schema instead of claimDetails (#248)

Approved after review in meeting TN-FMÞ-VH7-36 on the 9.10.2025.

thebingi committed:
Make expirationDate required (#243)

Approved after review in meeting TN-FMÞ-VH7-36 on the 9.10.2025.

thebingi committed:
Inconsistent parameter naming (#250)

On meeting TN-FMÞ-VH7-36 on the 9.10.2025, this PR was approved for being merged.

thebingi committed:
Set nullable fields correctly in claim schema (#245)

On meeting TN-FMÞ-VH7-36 on the 9.10.2025, this PR was approved for being merged.

thebingi committed:
Rename CollectionState -> CollectionStatus (#240)

On meeting TN-FMÞ-VH7-36 on the 9.10.2025, this PR was approved for being merged.

thebingi committed:
Feature/fix224 minor consistency (#239)

Meeting TN-FMÞ-VH7-36 on the 9.10.2025. The PR was approved for being merged.


* Minor consistency

* Minor consistency

* Name of endpoint

* An additional comment on the pull request.

kristinnstefansson committed:
Update links to stadlar.is in README.md

thebingi committed:
Feature/claims 3 1 improvements cleanup (#256)

Cleanup among other things for the linter:

* Spelling fix

* Corrected spelling and grammar.

* A little cleanup and polishing

* Adjustments to examples

* Minor changes

* Cleanup

* Patch Should return 200 and not 201

* Object type added in few places

* Path v.s Query issue

* Suggestions from Lint

* .redocly.yaml fix

kristinnstefansson committed:
Spelling fix in 315 variables

thebingi committed:
Update description for collection service (#257)

Helpful documentation tweak

kristinnstefansson committed:
Merge branch 'feature/claims_3_1_improvements' of https://github.com/stadlar/IST-FUT-FMTH into feature/claims_3_1_improvements

kristinnstefansson committed:
Cleanup of version numbers, change notes and first attempt at dates

kristinnstefansson committed:
Update claims specification version and correct references in the document

kristinnstefansson committed:
Update entry into force dates for ÍST TS 314 and TS 315 documents

asgeirhall commented:
Should the title not be updated?
title: IOBWS 3.0 Domestic Claims -> title: IOBWS 3.2 Domestic Claims

thebingi committed:
Final structural adjustments and schema cleanup (#258)

* Final structural adjustments and schema cleanup

* Required added to properties in dayAmountItem

* Fields added that can be used for filtering in GetClaims:
PaymentDate: This already exists in the claims pool, so it’s straightforward to add it as a filter option.
TemplateTransferDate: The date when a claim was moved to another template. This should also be relatively simple to add as a filter field. This is necessary for collection agencies to search by date, as it is currently more complicated since there is only a single endpoint.

Meeting TN-FMÞ-VH7-39 accepted this PR though there are adjustments that potentially will be accepted due to changes on RB functionality.

kristinnstefansson committed:
Cleanup and rename of files in preparation for publication

kristinnstefansson committed:
Updated dates for publication of 315

kristinnstefansson committed:
Update dates and publication information

kristinnstefansson committed:
Render 315 for publication
